File: MD.CMS.Administration/MD.CMS.Administration.Core.Web/scripts/businessLogic/dataAccess/entities/taxonomy.ts
/// <reference path="./base/iBaseEntity.ts" />
/// <reference path="./base/BaseEntity.ts" />
/// <reference path="./../../helpers.ts" />
/// <reference path="./taxonomyContent.ts" />
/// <reference path="./content.ts" />
namespace mdBusinessLogic.dataAccess.entities {
    export class taxonomy extends base.BaseEntity implements base.IBaseEntity<taxonomy>{
        public ParentId: number;
        public Name: string;
        public Description: string;
        public Parent: taxonomy;
        public Children: Array<taxonomy>;
        public Items: Array<taxonomyContent>;
        public FreeTextField: string;
        public Lcid: number;
        public FolderId: number;
        public TaxonomyPath: string;
        public Contents: Array<content>;
        public ParentArray: Array<taxonomy>;
        public ChildrenTotalCount: number;
        public ItemsTotalCount: number
        public Order: number;

        constructor(obj?: taxonomy) {
            super(obj);
            this.ParentId = 0;
            this.Name = '';
            this.Description = '';
            this.Parent = null;
            this.Children = new Array<taxonomy>();
            this.Items = new Array<taxonomyContent>();
            this.FreeTextField = '';
            this.Lcid = 0;
            this.FolderId = 0;
            this.TaxonomyPath = '';
            this.Contents = new Array<content>();
            this.ParentArray = new Array<taxonomy>();
            this.ChildrenTotalCount = 0;
            this.ItemsTotalCount = 0;
            this.Order = 0;
            if (obj != undefined && obj != null) {
                this.construct(obj);
            }
        }

        public construct(data: any) {
            super.construct(data);
            this.ParentId = this.getValue<number>(data, 'ParentId', 0);
            this.Name = this.getValue<string>(data, 'Name', '');
            this.Description = this.getValue<string>(data, 'Description', '');
            this.Parent = this.getValue<taxonomy>(data, 'Parent', new taxonomy());
            this.Children = this.getArrayConstructEntityValue<taxonomy>(data, 'Children', new Array<taxonomy>(), new taxonomy());
            this.Items = this.getArrayConstructEntityValue<taxonomyContent>(data, 'Items', new Array<taxonomyContent>(), new taxonomyContent());
            this.FreeTextField = this.getValue<string>(data, 'FreeTextField', '');
            this.Lcid = this.getValue<number>(data, 'Lcid', 0);
            this.FolderId = this.getValue<number>(data, 'FolderId', 0);
            this.TaxonomyPath = this.getValue<string>(data, 'TaxonomyPath', '');
            this.Contents = this.getArrayConstructEntityValue<content>(data, 'Contents', new Array<content>(), new content());
            this.ParentArray = helpers.loadParentArray(this, "Name", "TaxonomyPath");
            this.ChildrenTotalCount = this.getValue<number>(data, "ChildrenTotalCount", 0);
            this.ItemsTotalCount = this.getValue<number>(data, "ItemsTotalCount", 0);
            this.Order = this.getValue<number>(data, "Order", 0);
        }

        public clone(): taxonomy {
            return new taxonomy(this);
        }
    }
}

File: MD.CMS.BusinessLogic.Core/DataAccess/Entities/Taxonomy.cs
using System.Collections.Generic;
using MD.CMS.BusinessLogic.Core.DataAccess.Entities.Base;
using MD.CMS.BusinessLogic.Core.Helpers.Collections;
using MD.CMS.BusinessLogic.Core.Properties;
using MD.Tools.BaseDataAccess.Plugins.Core.Mapping.ReportDefinition.ReportDesigner.CustomAttributes;
namespace MD.CMS.BusinessLogic.Core.DataAccess.Entities
{
    public class Taxonomy : BaseHierarchycEntity<Taxonomy, long>, IHierarchyData
    {
        #region Attributes
        private long _parentId;
        private string _name;
        private string _description;
        private Taxonomy _parent;
        private List<Taxonomy> _children;
        private List<TaxonomyContent> _items;
        private List<Content> _contents;
        private string _freeTextField;
        private int _lcid;
        private int _folderId;
        private string _taxonomyPath;
        private int _childrenTotalCount;
        private int _itemsTotalCount;
        private int _order;
        #endregion

        #region Properties
        public int FolderId
        {
            get { return _folderId; }
            set { _folderId = value; }
        }

        public int LCID
        {
            get
            {
                if (_lcid.Equals(default(int)))
                    _lcid = Settings.Default.DefaultLcid;
                return _lcid;
            }
            set { _lcid = value; }
        }
        public string Name
        {
            get { return _name; }
            set { _name = value; }
        }

        public string Description
        {
            get { return _description; }
            set { _description = value; }
        }

        [OmitPropertyFromReport]
        public override Taxonomy Parent
        {
            get { return _parent; }
            set { _parent = value; }
        }

        public long ParentId
        {
            get { return _parentId; }
            set { _parentId = value; }
        }

        [OmitPropertyFromReport]
        public override List<Taxonomy> Children
        {
            get 
            {
                if (_children == null)
                {
                    _children = new List<Taxonomy>();
                }
                return _children; 
            }
            set { _children = value; }
        }

        [OmitPropertyFromReport]
        public List<Content> Contents
        {
            get { return _contents; }
            set { _contents = value; }
        }

        public bool IsNew
        {
            get
            {               
               return Id.Equals(default(long));              
            }
        }

        public List<TaxonomyContent> Items
        {
            get { return _items; }
            set { _items = value; }
        }

        public override string ToString()
        {
            return Id.ToString();
        }

        public string FreeTextField
        {
            get { return _freeTextField; }
            set { _freeTextField = value; }
        }

        public string TaxonomyPath
        {
            get { return _taxonomyPath; }
            set { _taxonomyPath = value; }
        }

        public int ChildrenTotalCount
        {
            get { return _childrenTotalCount; }
            set { _childrenTotalCount = value; }
        }

        public int ItemsTotalCount
        {
            get { return _itemsTotalCount; }
            set { _itemsTotalCount = value; }
        }

        public int Order { get => _order; set => _order = value; }
        #endregion
    }
}
